sentry-opentelemetry/lib/sentry/opentelemetry/span_processor.rb

```diff
@@ -0,0 +1,159 @@
+require 'singleton'
+
+module Sentry
+  module OpenTelemetry
+    TraceData = Struct.new(:trace_id, :span_id, :parent_span_id, :parent_sampled, :baggage)
+
+    class SpanProcessor < ::OpenTelemetry::SDK::Trace::SpanProcessor
+      include Singleton
+
+      SEMANTIC_CONVENTIONS = ::OpenTelemetry::SemanticConventions::Trace
+
+      # The mapping from otel span ids to sentry spans
+      # @return [Hash]
+      attr_reader :span_map
+
+      def initialize
+        @span_map = {}
+      end
+
+      def on_start(otel_span, parent_context)
+        return unless Sentry.initialized? && Sentry.configuration.instrumenter == :otel
+        return unless otel_span.context.valid?
+        return if from_sentry_sdk?(otel_span)
+
+        trace_data = get_trace_data(otel_span, parent_context)
+
+        sentry_parent_span = @span_map[trace_data.parent_span_id] if trace_data.parent_span_id
+
+        sentry_span = if sentry_parent_span
```

Comment:
the root span from otel becomes a transaction and all other child spans become child spans of the transaction in sentry